STL_Extension/include/CGAL/assertions.h

```diff
@@ -100,7 +100,7 @@ inline bool possibly(Uncertain<bool> c);
 #else // no CGAL_NO_ASSERTIONS
 #  define CGAL_assertion(EX) \
    (CGAL::possibly(EX)?(static_cast<void>(0)): ::CGAL::assertion_fail( # EX , __FILE__, __LINE__))
-#  if __cpp_lib_uncaught_exceptions // C++17
+#  if defined( __cpp_lib_uncaught_exceptions) || _MSVC_LANG >= 201103L // C++17
```

Comment:
That line is wrong for two reasons:
1. `#if __cpp_lib_uncaught_exceptions` and `if defined( __cpp_lib_uncaught_exceptions)` do not test the same thing, and in particular it differs if `__cpp_lib_uncaught_exceptions` is defined to `0`.
2. `std::uncaught_exceptions` is from C++17. So `_MSVC_LANG` must be compared to `201703L`.

The right line is:
```
#  if __cpp_lib_uncaught_exceptions || ( _MSVC_LANG >= 201703L )
```